arangod/VocBase/vocbase.cpp

```diff
@@ -575,13 +575,17 @@ std::shared_ptr<arangodb::LogicalCollection> TRI_vocbase_t::createCollectionWork
   auto collection =
       std::make_shared<arangodb::LogicalCollection>(*this, parameters, false);
 
+  return persistCollection(collection);
+}
+
+std::shared_ptr<arangodb::LogicalCollection> TRI_vocbase_t::persistCollection(std::shared_ptr<arangodb::LogicalCollection>& collection){
```

Comment:
Why does this method return?
I cannot modify the shared_ptr anyways.
So can we just make it a `void` or better `Result` method?
The input collection should than be a const shared_ptr,
Or even a `LogicalCollection const&`

Reply:
Ah okay in the hidden section there is a registerCollection which will e require a shared_ptr input, so we need to keep this, but we can improve the return type. to none.

Reply:
And after it's being registered it will be persisted and returned :P